### ISSUE TITLE
[Realtime DB] update data by key creates an array of chars instead of string

### ISSUE THREAD

**Issue by NickIliev** [EddyVerbruggen/nativescript-plugin-firebase#257]
Let's assume we have created the following entry
```
export function setValue() {
    // to store an array of JSON objects
    firebase.setValue(
        '/owner',
        { name: "Nick Iliev" }  // query will return {"name":"Nick Iliev"}
    ).then(() => {
        console.log("value set!")
    })
}
```
I can add new key by update like this
```
export function updateValue() {
    firebase.update(
        '/owner',
        { name: "Nikolay P. Iliev" }
    ).then(() => {
        console.log("value updated!");
    })
}
```

What can be done wth update is to modify an existing key like this
```
export function updateValueByKey() {
    firebase.update(
        '/owner/name',
        "Native Nick"
    ).then(() => {
        console.log("[name] value updated!");
    })
}
```

This works BUT not as expected as instead of `{"name":"Native Nick","address":"Sofia, Bulgaria"}`
I am receiving 
```
{"name":["N","a","t","i","v","e"," ","N","i","c","k"],"address":"Sofia, Bulgaria"}
```

Ideas on how to handle that behaviour? it would be great to support this method as it is easier to update deep nested key-value pairs.

**Comment by EddyVerbruggen** [comment]
Would indeed be nice to be able to do that, is this iOS, Android, or both?

**Comment by NickIliev** [comment]
Observed in Android app - will enable iOS one of these days and report back if the same occurs there

**Comment by EddyVerbruggen** [comment]
Turned out that on iOS there wasn't any data saved at all with the same code.

So the native SDK expects a Map, not a String, that's where things went south. So I'm changing the code to check if the thing passed to `update` is not an object I'm taking the last part of the path and use that as the key for a Map, and the value as the Map's value.

Tested with your scenario and it works fine, also still works fine with objects being passed in.

Would you mind doing a little code review of [this commit](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/commit/a7f8d8c16e974186c5e40188458b40461df7f4ec)? Could be a little cleaner by using the common superclass, but hey, nobody's perfect :)

**Comment by EddyVerbruggen** [comment]
Released with 3.9.1.
